app/components/Menu/MenuDropdown.tsx

```diff
@@ -0,0 +1,102 @@
+import { View, type ViewStyle } from "react-native"
+import { useRef, useMemo, memo } from "react"
+import { themed } from "../../theme/theme"
+import { Portal } from "../Portal"
+import { MenuDropdownItem } from "./MenuDropdownItem"
+import { useSubmenuState } from "./useSubmenuState"
+import { menuSettings } from "./menuSettings"
+import { type Position, type DropdownMenuItem, type MenuItem, MENU_SEPARATOR } from "./types"
+import { getUUID } from "../../utils/random/getUUID"
+import { Separator } from "../Separator"
+
+interface MenuDropdownProps {
+  items: (DropdownMenuItem | typeof MENU_SEPARATOR)[]
+  position: Position
+  onItemPress: (item: MenuItem) => void
+  isSubmenu?: boolean
+}
+
+const MenuDropdownComponent = ({ items, position, onItemPress, isSubmenu }: MenuDropdownProps) => {
+  const portalName = useRef(`${isSubmenu ? "submenu" : "dropdown"}-${getUUID()}`).current
+  const { openSubmenu, submenuPosition, handleItemHover } = useSubmenuState(position)
+
+  const isSeparator = (item: MenuItem | typeof MENU_SEPARATOR): item is typeof MENU_SEPARATOR => {
+    return item === MENU_SEPARATOR
+  }
```

Comment:
I'd move this function out of the component, since it doesn't seem to rely on anything in scope here in the component. Could either move it below the component as a hoisted `function isSeparator` or into a util somewhere.